client/state/selectors/is-private-site.js

```diff
@@ -12,9 +13,17 @@ import { getRawSite } from 'state/sites/selectors';
  */
 export default function isPrivateSite( state, siteId ) {
 	const site = getRawSite( state, siteId );
-	if ( ! site ) {
-		return null;
+
+	if ( site ) {
+		return site.is_private;
+	}
+
+	const settings = getSiteSettings( state, siteId );
+
+	if ( settings ) {
+		// Site settings returns a numerical value for blog_public.
+		return parseInt( settings.blog_public, 10 ) !== 1;
```

Comment:
Can we rely on `settings.blog_public` to always be a string? My hunch is maybe not (via optimistic state updates), in which case what you have here is probably good for safety. If we could be confident, we might be able to simply `return '1' !== settings.blog_public`.